mmv1/products/vmwareengine/PrivateCloud.yaml

```diff
@@ -89,6 +91,17 @@ parameters:
     url_param_only: true
     description: |
       The ID of the PrivateCloud.
+  - !ruby/object:Api::Type::NestedObject
+    name: 'deletionParameters'
```

Comment:
I initially planned to add only the `deletion_delay_hours` parameter. However, this made me unable to achieve the following behavior:
- if user enters no value, pass the delete API url without the delay hr mask. This would add the default delay hour which is set at the api level. We dont want to hardcode this value at the terraform level so that it can be easily modified and is at one place. 
- if the user adds any value (0 and non zero), use this value for the delay hr mask in the delete url.  

Since Terraform cant distinguish between empty and zero values, I created a nested object so that I can check its length to find out if the user did set the `deletion_delay_hours` value.  Please let me know if you feel there is any other better approach to handle this use case.

Reply:
We don't have a strong pattern for this kind of situation but we usually use a second boolean field to control behavior in some way. For example:

- `send_deletion_delay_hours_if_zero`: similar to https://github.com/GoogleCloudPlatform/magic-modules/pull/10717. It would mean that people who want to send zero need to really really mean it!
- you could also do a field like `use_default_delay_hours` to indicate that the server should by default always send whatever is set (including zero) unless the field is set.

I'm guessing you'd prefer something like the former since folks are less likely to set `0` for delay_hours accidentally.

Reply:
I am concerned that introducing an additional boolean field alongside the delay hours field could potentially introduce user error, particularly given its absence from the VMware Engine API.

Creating a nested object looks like a cleaner and more adaptive approach. This would also hold additional delete API parameters which might be added in the near future.

Wdyt?

Reply:
Used `send_deletion_delay_hours_if_zero` since virtual fields dont support nested fields at the moment